pkg/controller/volume/attachdetach/attach_detach_controller_test.go

```diff
@@ -161,17 +161,15 @@ func attachDetachRecoveryTestCase(t *testing.T, extraPods1 []*v1.Pod, extraPods2
 	}
 
 	for _, pod := range pods.Items {
-		podToAdd := pod
-		podInformer.GetIndexer().Add(&podToAdd)
+		podInformer.GetIndexer().Add(&pod)
```

Comment:
Sometimes you can have issues when passing the pointer from a variable defined in the for loop because each for loop iteration will reuse the same variable instead of creating a new one each time.  Can you verify that this Add() call will actually copy the contents of the pointer, instead of just storing the pointer value?